src/bandersnatch/package.py

```diff
@@ -1,91 +1,56 @@
 import asyncio
-import hashlib
-import html
 import logging
-import sys
-from json import dump
-from pathlib import Path
-from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
-from urllib.parse import unquote, urlparse
+from typing import Dict, List
 
 from packaging.utils import canonicalize_name
 
-from . import utils
-from .master import PackageNotFound, StalePage
+from .errors import PackageNotFound, StaleMetadata
+from .master import Master, StalePage
 
 from .filter import filter_metadata_plugins  # isort:skip
 from .filter import filter_release_file_plugins  # isort:skip
 from .filter import filter_release_plugins  # isort:skip
 
-
-if TYPE_CHECKING:  # pragma: no cover
-    from .mirror import Mirror
-
-
 # Bool to help us not spam the logs with certain log messages
 display_filter_log = True
 logger = logging.getLogger(__name__)
 
 
-class StaleMetadata(Exception):
-    """We attempted to retreive metadata from PyPI, but it was stale."""
-
-    def __init__(self, package_name: str, attempts: int) -> None:
-        super().__init__()
-        self.package_name = package_name
-        self.attempts = attempts
-
-    def __str__(self) -> str:
-        return f"Stale serial for {self.package_name} after {self.attempts} attempts"
-
-
 class Package:
-    def __init__(self, name: str, serial: Union[int, str], mirror: "Mirror",) -> None:
+    def __init__(self, name: str, serial: int = 0) -> None:
         self.name = canonicalize_name(name)
         self.raw_name = name
         self.serial = serial
-        self.mirror = mirror
+
+        self._metadata = None
 
     @property
-    def json_file(self) -> Path:
-        return Path(self.mirror.webdir / "json" / self.name)
+    def metadata(self) -> dict:
```

Comment:
```suggestion
    def metadata(self) -> Dict:
```

NIT: Don't type with standard typing objects - Using Generics types is preferred from the typing module: https://docs.python.org/3/library/typing.html#generics